### ISSUE TITLE
[Issue]: DNLA Casting to Samsung TV Broken when HTTPS Enabled 

### ISSUE THREAD

**Issue by worky68** [jellyfin/jellyfin#9540]
### Please describe your bug

Some context:

 I've had no issues using DNA Casting to my Samsung TVs since installing Jellyfin other than some DNLA profile adjustments to account for HEVC video. 

The issue:

I recently to enable HTTPS in Jellyfin using a self signed pfx. Note that HTTPS is simply enabled, it is not required. I noticed that I was no longer able to use the DNLA Cast feature to my Samsung TV. From the TV side nothing occurs and its like I did nothing. On the Jellyfin client side nothing seems to visually occur as well. However an error does get logged (see below).

Between enabling HTTPS and attempting to use DNLA Cast was some time so I did not immediately realized it could be that I enabled HTTPS. I have 2 Samsung TVs that I tested with during my investigation of the issue. I discovered that the issue seems to be that Jellyfin is passing HTTPS URLs when capturing traffic between my server and the Samsung TVs. During troubleshooting, I had looked at previous issues reported related to DNLA; The understanding I came to is that DNLA devices often have issues using HTTPS thus Jellyfin would use HTTP instead of HTTPS as long as HTTPS wasn't required. As you can see in my tcpdump captures, the URLs being passed to my TV are HTTPS when it is not required within the network settings. 

My confirmation that is it at least related to enabling HTTPS is that I disabled HTTPS and restarted Jellyfin. After which, the DNLA Cast to my Samsung TVs was working once again.

One final test I did was to access Jellyfin via the HTTP URL within a browser and try to DNLA Cast to my Samsung TV. I was hoping it would use the URL that I was currently accessing Jellyfin with. This did not work and the same error occurred.

Side note:

I would expect HTTPS to fail with a self signed certificate anyways. It would make sense to allow setting which URL to use within the DNLA menu.

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: Debian (Server), Windows (Clients)
- Virtualization: None
- Clients: Browser, Jellyfin Media Player 1.9
- Browser: Firefox 111.0.1
```

### Jellyfin logs

```shell
Jellyfin logging the connection to the TV (this seems to occur upon started up not after I connect in the Cast menu):

[2023-03-26 00:00:55.030 -04:00] [INF] DLNA Session created for "[TV] Samsung 6 Series (70)" - "UN70NU6070"

Relevant error in the Jellyfin log when clicking play on a video:

[2023-03-26 00:43:53.277 -04:00] [ERR] Error processing request. URL "POST" "/Sessions/8f4d107d2bb3a8de3f1391dfbd97cdae/Playing".
System.Net.Http.HttpRequestException: Response status code does not indicate success: 500 (Internal Server Error).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at Emby.Dlna.PlayTo.SsdpHttpClient.SendCommandAsync(String baseUrl, DeviceService service, String command, String postData, String header, CancellationToken cancellationToken)
   at Emby.Dlna.PlayTo.Device.SetAvTransport(String url, String header, String metaData, CancellationToken cancellationToken)
   at Emby.Dlna.PlayTo.PlayToController.SetPlaylistIndex(Int32 index, CancellationToken cancellationToken)
   at Emby.Dlna.PlayTo.PlayToController.PlayItems(IEnumerable`1 items, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Session.SessionManager.SendMessageToSession[T](SessionInfo session, SessionMessageType name, T data, CancellationToken cancellationToken)
   at Emby.Server.Implementations.Session.SessionManager.SendPlayCommand(String controllingSessionId, String sessionId, PlayRequest command, CancellationToken cancellationToken)
   at Jellyfin.Api.Controllers.SessionController.Play(String sessionId, PlayCommand playCommand, Guid[] itemIds, Nullable`1 startPositionTicks, String mediaSourceId, Nullable`1 audioStreamIndex, Nullable`1 subtitleStreamIndex, Nullable`1 startIndex)
   at lambda_method1289(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

### Please attach any browser or client logs here

I didn't see any logging of the DNLA requests or responses in the jellyfin log even with debug logging enabled and restarting Jellyfin. I decided to use tcpdump to get captures of a DNLA request and response between the server and the Samsung TV. This just is one of the interactions between the sever and the Samsung TV after pressing play on a video with the Server connected to the TV via DNLA.

Sent from the Server to the TV:

0x0000:  4500 01dc 5d93 4000 4006 59d1 c0a8 0038  E...].@.@.Y....8
0x0010:  c0a8 002f b966 23ed 2b5a 164b b6b8 48a7  .../.f#.+Z.K..H.
0x0020:  8018 01f6 8386 0000 0101 080a 680d 7e68  ............h.~h
0x0030:  001f 967d 636f 6c49 6e66 6f3d 2671 756f  ...}colInfo=&quo
0x0040:  743b 6874 7470 2d67 6574 3a2a 3a69 6d61  t;http-get:*:ima
0x0050:  6765 2f6a 7065 673a 444c 4e41 2e4f 5247  ge/jpeg:DLNA.ORG
0x0060:  5f50 4e3d 4a50 4547 5f54 4e3b 444c 4e41  _PN=JPEG_TN;DLNA
0x0070:  2e4f 5247 5f4f 503d 3030 3b44 4c4e 412e  .ORG_OP=00;DLNA.
0x0080:  4f52 475f 4349 3d30 3b44 4c4e 412e 4f52  ORG_CI=0;DLNA.OR
0x0090:  475f 464c 4147 533d 3030 4430 3030 3030  G_FLAGS=00D00000
0x00a0:  3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
0x00b0:  3030 3030 3030 3030 2671 756f 743b 2072  00000000&quot;.r
0x00c0:  6573 6f6c 7574 696f 6e3d 2671 756f 743b  esolution=&quot;
0x00d0:  3136 3078 3930 2671 756f 743b 2667 743b  160x90&quot;&gt;
0x00e0:  6874 7470 733a 2f2f 3139 322e 3136 382e  https://192.168.
0x00f0:  302e 3536 3a38 3932 302f 4974 656d 732f  0.56:8920/Items/
0x0100:  3336 3238 6466 3130 3438 3637 3166 3662  3628df1048671f6b
0x0110:  6135 6437 3165 3930 3561 6339 3764 3066  a5d71e905ac97d0f
0x0120:  2f49 6d61 6765 732f 5072 696d 6172 792f  /Images/Primary/
0x0130:  302f 3430 3339 3466 3338 3931 3231 3833  0/40394f38912183
0x0140:  3538 6536 6134 6339 3437 6463 6433 3731  58e6a4c947dcd371
0x0150:  3532 2f6a 7067 2f31 3630 2f31 3630 2f30  52/jpg/160/160/0
0x0160:  2f30 266c 743b 2f72 6573 2667 743b 266c  /0&lt;/res&gt;&l
0x0170:  743b 2f69 7465 6d26 6774 3b26 6c74 3b2f  t;/item&gt;&lt;/
0x0180:  4449 444c 2d4c 6974 6526 6774 3b3c 2f43  DIDL-Lite&gt;</C
0x0190:  7572 7265 6e74 5552 494d 6574 6144 6174  urrentURIMetaDat
0x01a0:  613e 3c2f 6d3a 5365 7441 5654 7261 6e73  a></m:SetAVTrans
0x01b0:  706f 7274 5552 493e 3c2f 534f 4150 2d45  portURI></SOAP-E
0x01c0:  4e56 3a42 6f64 793e 3c2f 534f 4150 2d45  NV:Body></SOAP-E
0x01d0:  4e56 3a45 6e76 656c 6f70 653e            NV:Envelope>

Sent to the Server from the TV:

0x0000:  4500 02ea dede 4000 4006 d777 c0a8 002f  E.....@.@..w.../
0x0010:  c0a8 0038 23ed b966 b6b8 48a7 2b5a 17f3  ...8#..f..H.+Z..
0x0020:  8018 013d b016 0000 0101 080a 001f 96b5  ...=............
0x0030:  680d 7e68 4854 5450 2f31 2e31 2035 3030  h.~hHTTP/1.1.500
0x0040:  2049 6e74 6572 6e61 6c20 5365 7276 6572  .Internal.Server
0x0050:  2045 7272 6f72 0d0a 4558 543a 200d 0a43  .Error..EXT:...C
0x0060:  4f4e 4e45 4354 494f 4e3a 2063 6c6f 7365  ONNECTION:.close
0x0070:  0d0a 434f 4e54 454e 542d 5459 5045 3a20  ..CONTENT-TYPE:.
0x0080:  7465 7874 2f78 6d6c 3b20 6368 6172 7365  text/xml;.charse
0x0090:  743d 2275 7466 2d38 220d 0a53 4552 5645  t="utf-8"..SERVE
0x00a0:  523a 5550 6e50 2f31 2e30 2c20 5361 6d73  R:UPnP/1.0,.Sams
0x00b0:  756e 6720 416c 6c53 6861 7265 2053 6572  ung.AllShare.Ser
0x00c0:  7665 722f 312e 300d 0a44 4154 453a 2053  ver/1.0..DATE:.S
0x00d0:  756e 2c20 3236 204d 6172 2032 3032 3320  un,.26.Mar.2023.
0x00e0:  3035 3a33 313a 3535 2047 4d54 0d0a 5553  05:31:55.GMT..US
0x00f0:  4552 2d41 4745 4e54 3a20 5361 6d73 756e  ER-AGENT:.Samsun
0x0100:  6720 5573 6572 2d41 6765 6e74 2044 4c4e  g.User-Agent.DLN
0x0110:  4144 4f43 2f31 2e35 300d 0a43 6f6e 7465  ADOC/1.50..Conte
0x0120:  6e74 2d4c 656e 6774 683a 2034 3430 0d0a  nt-Length:.440..
0x0130:  0d0a 3c3f 786d 6c20 7665 7273 696f 6e3d  ..<?xml.version=
0x0140:  2231 2e30 2220 656e 636f 6469 6e67 3d22  "1.0".encoding="
0x0150:  7574 662d 3822 3f3e 3c73 3a45 6e76 656c  utf-8"?><s:Envel
0x0160:  6f70 6520 733a 656e 636f 6469 6e67 5374  ope.s:encodingSt
0x0170:  796c 653d 2268 7474 703a 2f2f 7363 6865  yle="http://sche
0x0180:  6d61 732e 786d 6c73 6f61 702e 6f72 672f  mas.xmlsoap.org/
0x0190:  736f 6170 2f65 6e63 6f64 696e 672f 2220  soap/encoding/".
0x01a0:  786d 6c6e 733a 733d 2268 7474 703a 2f2f  xmlns:s="http://
0x01b0:  7363 6865 6d61 732e 786d 6c73 6f61 702e  schemas.xmlsoap.
0x01c0:  6f72 672f 736f 6170 2f65 6e76 656c 6f70  org/soap/envelop
0x01d0:  652f 223e 3c73 3a42 6f64 793e 3c73 3a46  e/"><s:Body><s:F
0x01e0:  6175 6c74 3e3c 6661 756c 7463 6f64 653e  ault><faultcode>
0x01f0:  733a 436c 6965 6e74 3c2f 6661 756c 7463  s:Client</faultc
0x0200:  6f64 653e 3c66 6175 6c74 7374 7269 6e67  ode><faultstring
0x0210:  3e55 506e 5045 7272 6f72 3c2f 6661 756c  >UPnPError</faul
0x0220:  7473 7472 696e 673e 3c64 6574 6169 6c3e  tstring><detail>
0x0230:  3c55 506e 5045 7272 6f72 2078 6d6c 6e73  <UPnPError.xmlns
0x0240:  3d22 7572 6e3a 7363 6865 6d61 732d 7570  ="urn:schemas-up
0x0250:  6e70 2d6f 7267 3a63 6f6e 7472 6f6c 2d31  np-org:control-1
0x0260:  2d30 223e 3c65 7272 6f72 436f 6465 3e37  -0"><errorCode>7
0x0270:  3136 3c2f 6572 726f 7243 6f64 653e 3c65  16</errorCode><e
0x0280:  7272 6f72 4465 7363 7269 7074 696f 6e3e  rrorDescription>
0x0290:  5265 736f 7572 6365 206e 6f74 2066 6f75  Resource.not.fou
0x02a0:  6e64 3c2f 6572 726f 7244 6573 6372 6970  nd</errorDescrip
0x02b0:  7469 6f6e 3e3c 2f55 506e 5045 7272 6f72  tion></UPnPError
0x02c0:  3e3c 2f64 6574 6169 6c3e 3c2f 733a 4661  ></detail></s:Fa
0x02d0:  756c 743e 3c2f 733a 426f 6479 3e3c 2f73  ult></s:Body></s
0x02e0:  3a45 6e76 656c 6f70 653e                 :Envelope>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by slopsjon** [comment]
I had the same problem. If I selected https and force https you don't see the TV as available to cast to. If I remove the force https then I can select the TV but nothing happens when I press play. I've gone back to http for now.

**Comment by seaeagle1** [comment]
I've just struggled with the same issue: Jellyfin automatically uses the HTTPS port in the PlayTo function when HTTPS is enabled. My mediaplayer didn't understand all the SSL stuff and just didn't start playing.

Disabling HTTPS should work, and for myself I've now redirected the HTTPS port to the HTTP port in my firewall for the mediaplayer.

**Comment by Shadowghost** [comment]
Might be fixed by #8147
You can also force URL overrides for specific subnets or Adresse

**Comment by slopsjon** [comment]
I managed to find a work around or rather another way of creating the ssl cert by following this guys way of doing it. I used duckdns and pihole for the cnames. [https://www.youtube.com/watch?v=qlcVx-k-02E](url) I don't get any warning this way about not trusted certificates. I didn't turn ssl on in jellyfin as it wants a local cert.
